### ISSUE TITLE
Fences have no collision check when connecting

### ISSUE THREAD

**Issue by Moonyol** [PixelGuys/Cubyz#854]
Fences do not check whether you are would be inside them if they connected which makes you go inside them and forces you up so this can happen
https://github.com/user-attachments/assets/d4b6b104-118b-4435-a180-8ed0f9933eaa

**Comment by IntegratedQuantum** [comment]
I think the problem is that it only checks the hitbox of the block placed against the player hitbox, without applying the changes based on the neighbor blocks.

**Comment by IntegratedQuantum** [comment]
As we recently discovered, this allows you to make a fast elevator that teleports you upwards.

**Comment by Moonyol** [comment]
Is that similar to the video? I stood next to a fence underground and placed another next to it.

**Comment by IntegratedQuantum** [comment]
Yes, except it actually only needs a single block column if know what you are doing.
